lib/dslash_quda.cu

```diff
@@ -123,11 +151,11 @@ namespace quda {
       dslashStart[i] = qudaEventCreate();
     }
 #ifdef NVSHMEM_COMMS
-    sync_arr = static_cast<shmem_sync_t *>(device_comms_pinned_malloc(2 * QUDA_MAX_DIM * sizeof(shmem_sync_t)));
```

Comment:
why the jump from 2 to 4?

Reply:
This is to accomodate the dslash and exchange ghost counters, i.e. dims * dirs *2 (last 2 for dslash / exchange host)